privacyidea/lib/policydecorators.py

```diff
@@ -192,8 +192,12 @@ def auth_cache(wrapped_function, user_object, passw, options=None):
                 g.audit_object.add_policy(next(iter(auth_cache_dict.values())))
                 return True, {"message": "Authenticated by AuthCache."}
 
-    # If nothing else returned, we return the wrapped function
-    return wrapped_function(user_object, passw, options)
+    # If nothing else returned, call the wrapped function
+    res, reply_dict = wrapped_function(user_object, passw, options)
+    if res:
+        # If authentication is successful, we store the password in auth_cache
+        add_to_cache(user_object.login, user_object.realm, user_object.resolver, passw)
```

Comment:
shouldn't we check here if the auth_cache policy is active? Otherwise we would add any authentication to the cache, or am i missing something?

Reply:
Oups, I think you are right. Will add a test and change this.

Reply:
done in ef003805